general/freetiersetup/freetiersetup.go

```diff
@@ -0,0 +1,164 @@
+package freetiersetup
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jfrog/jfrog-cli-core/v2/common/commands"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-client-go/http/httpclient"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	ioUtils "github.com/jfrog/jfrog-client-go/utils/io"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+
+	"github.com/pkg/browser"
+)
+
+const (
+	myJfrogEndPoint          = "https://myjfrog-api.jfrog.info/api/v1/activation/cloud/cli/getStatus/"
+	defaultSyncSleepInterval = 5 * time.Second  // 5 seconds
+	maxWaitMinutes           = 10 * time.Minute // 10 minutes
+)
+
+type FreeTierSetupCommand struct {
+	registrationURL string
+	id              uuid.UUID
+	serverDetails   *config.ServerDetails
+	progress        ioUtils.ProgressMgr
+}
+
+func (ftc *FreeTierSetupCommand) ServerDetails() (*config.ServerDetails, error) {
+	return ftc.serverDetails, nil
+}
+
+func (ftc *FreeTierSetupCommand) SetProgress(progress ioUtils.ProgressMgr) {
+	ftc.progress = progress
+}
+
+func (ftc *FreeTierSetupCommand) IsFileProgress() bool {
+	return false
+}
+
+func NewFreeTierSetupCommand(url string) *FreeTierSetupCommand {
+	return &FreeTierSetupCommand{
+		registrationURL: url,
+		id:              uuid.New(),
+	}
+}
+
+func (ftc *FreeTierSetupCommand) Run() (err error) {
+	browser.OpenURL(ftc.registrationURL + "?id=" + ftc.id.String())
+	ftc.progress.SetHeadlineMsg("Welcome to JFrog CLI! To complete your JFrog environment setup, please fill out the details in your browser")
+	server, err := ftc.getServerDetails()
+	if err != nil {
+		return
+	}
+	err = configServer(server)
+	if err != nil {
+		return err
+	}
+	println("Your new JFrog environment is ready!")
+	println("	1.CD into your code project directory")
+	println("	2.Run \"jf project init\"")
+	println("	3.Read more about how to get started at https://...")
+	return nil
+}
+
+func (ftc *FreeTierSetupCommand) CommandName() string {
+	return "setup"
+}
+
+func (ftc *FreeTierSetupCommand) getServerDetails() (serverDetails *config.ServerDetails, err error) {
```

Comment:
Since all commands already have a method named ```getServerDetails``` which is used for a different purpose than this method, to avoid confusion, I suggest renaming this method to ```getNewServerDetails```.